autoblocks/_impl/testing/run.py

```diff
@@ -161,10 +127,22 @@ async def run_test_case_unsafe(
     """
     async with test_case_semaphore_registry[test_id]:
         if inspect.iscoroutinefunction(fn):
-            return await fn(test_case)
+            output = await fn(test_case)
         else:
             ctx = contextvars.copy_context()
-            return await global_state.event_loop().run_in_executor(None, ctx.run, fn, test_case)
+            output = await global_state.event_loop().run_in_executor(None, ctx.run, fn, test_case)
+
+    await global_state.http_client().post(
+        f"{cli()}/results",
+        json=dict(
+            testExternalId=test_id,
+            testCaseHash=test_case._cached_hash,
+            testCaseBody=serialize_test_case(test_case),
+            testCaseOutput=serialize(output),
+        ),
```

Comment:
the issue is that this part was not in the `_unsafe` method but was being called afterward where we have no outer error handler sending errors, so this was being swallowed by `all_settled`, where we don't do anything w/ the results

note that we're not try-catching the POST request here, since the client already suppresses those (those errors are handled and displayed by the CLI). here we are making sure we are try-catching the building of the JSON payload, which is where the error was occurring